cmd/access/access_node_builder.go

```diff
@@ -0,0 +1,200 @@
+	}
+
+	// and also the unstaked bind address
+	if anb.unstakedNetworkBindAddr == cmd.NotSet {
+		anb.Logger.Fatal().Msg("unstaked bind address not set")
+	}
+}
+
+func (anb *AccessNodeBuilder) EnqueueUnstakedNetworkInit() {
+	anb.Component("unstaked network", func(node *cmd.FlowNodeBuilder) (module.ReadyDoneAware, error) {
+		codec := jsoncodec.NewCodec()
+
+		// setup the Ping provider to return the software version and the finalized block height
+		pingProvider := p2p.PingInfoProviderImpl{
+			SoftwareVersionFun: func() string {
+				return build.Semver()
+			},
+			SealedBlockHeightFun: func() (uint64, error) {
+				head, err := node.State.Sealed().Head()
+				if err != nil {
+					return 0, err
+				}
+				return head.Height, nil
+			},
+		}
+
+		libP2PNodeFactory, err := p2p.DefaultLibP2PNodeFactory(node.Logger.Level(zerolog.ErrorLevel),
+			node.Me.NodeID(),
+			anb.unstakedNetworkBindAddr,
+			node.NetworkKey,
+			node.RootBlock.ID().String(),
+			p2p.DefaultMaxPubSubMsgSize,
+			node.Metrics.Network,
+			pingProvider)
+		if err != nil {
+			return nil, fmt.Errorf("could not generate libp2p node factory: %w", err)
+		}
+
+		peerUpdateInterval := time.Hour // pretty much not needed for the unstaked access node
+
+		var msgValidators []network.MessageValidator
+		if anb.staked {
+			msgValidators = p2p.DefaultValidators(node.Logger, node.Me.NodeID())
+		} else {
+			// for an unstaked node, use message sender validator but not target validator since the staked AN will
+			// be broadcasting messages to ALL unstaked ANs without knowing their target IDs
+			msgValidators = []network.MessageValidator{
+				// filter out messages sent by this node itself
+				validator.NewSenderValidator(node.Me.NodeID()),
+				// but retain all the 1-k messages even if they are not intended for this node
+			}
+		}
+
+		anb.unstakedMiddleware = p2p.NewMiddleware(node.Logger.Level(zerolog.ErrorLevel),
+			libP2PNodeFactory,
+			node.Me.NodeID(),
+			node.Metrics.Network,
+			node.RootBlock.ID().String(),
+			peerUpdateInterval,
+			p2p.DefaultUnicastTimeout,
+			msgValidators...)
+
+		participants, err := node.State.Final().Identities(p2p.NetworkingSetFilter)
+		if err != nil {
+			return nil, fmt.Errorf("could not get network identities: %w", err)
+		}
+
+		// creates topology, topology manager, and subscription managers
+		//
+		// topology
+		// subscription manager
+		subscriptionManager := p2p.NewChannelSubscriptionManager(anb.unstakedMiddleware)
+		var top network.Topology
+		if anb.staked {
+			top = topology.EmptyListTopology{}
+		} else {
+			upstreamANIdentifier, err := flow.HexStringToIdentifier(anb.stakedAccessNodeIDHex)
+			if err != nil {
+				return nil, fmt.Errorf("failed to convert node id string %s to Flow Identifier: %v", anb.stakedAccessNodeIDHex, err)
+			}
+			top = topology.NewFixedListTopology(upstreamANIdentifier)
+		}
+
+		// creates network instance
+		net, err := p2p.NewNetwork(node.Logger,
+			codec,
+			participants,
```

Comment:
Is this correct? `participants` is the list of staked nodes, right? Does it make sense to use these as the participants for the unstaked network?

Reply:
👍 
More exactly: are we sure that, even if it wanted, the unstaked node cannot access staked network information (see my comment on the PR on having a != bootstrap config)